addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -946,7 +946,11 @@ const MultiUserValueWidget = UserValueWidget.extend({
      */
     getValue: function (methodName) {
         const value = this._userValueWidgets.map(widget => {
-            return widget.getValue(methodName);
+            let widgetValue = widget.getValue(methodName);
+            if (widget.$el.hasClass('o_we_so_color_palette')) {
+                widgetValue = ColorpickerDialog.isCSSColor(widgetValue) ? widgetValue : `var(--${widgetValue})`;
+            }
+            return widgetValue;
```

Comment:
Maybe it should be an option on the colorpicker to automatically return var(--xxx) in that case. Would be better than a hack in we-multi to make it work I think.